napari/plugins/_builtins.py

```diff
@@ -23,4 +23,5 @@ def napari_get_reader(path: str):
     callable
         function that returns layer_data to be handed to viewer._add_layer_data
     """
-    return lambda path: [(None, {'path': path})]
+    if path.endswith(('tif', 'png', 'jpg')):
```

Comment:
mostly just a placeholder... see my main review comment about handling "fallback behavior".  If we _do_ want to implement the "when-all-else-fails" behavior as our own internal plugin, then we would probably remove any conditional here and return to the way it is in `master` where this function simply returns `lambda path: [(None, {'path': path})]` (which just tells `add_layer_from_data` to call `add_image(path=path)`).

The alternative is to let `get_layer_data_from_plugins` just return nothing, and put the fallback logic (which would basically be our current `io.magic_read` logic) in line 95 in `add_layers_mixin` where the `print()` statement currently is.